x-pack/plugins/security_solution/public/detections/components/alerts_table/actions.tsx

```diff
@@ -157,59 +161,33 @@ export const getThresholdAggregationData = (ecsData: Ecs | Ecs[]): ThresholdAggr
   const thresholdEcsData: Ecs[] = Array.isArray(ecsData) ? ecsData : [ecsData];
   return thresholdEcsData.reduce<ThresholdAggregationData>(
     (outerAcc, thresholdData) => {
-      const threshold = thresholdData.signal?.rule?.threshold as string[];
+      const threshold =
+        getField(thresholdData, ALERT_RULE_PARAMETERS).threshold ??
+        thresholdData.signal?.rule?.threshold;
 
-      let aggField: string[] = [];
-      let thresholdResult: {
-        terms?: Array<{
-          field?: string;
+      const thresholdResult: {
+        terms: Array<{
+          field: string;
           value: string;
         }>;
         count: number;
         from: string;
-      };
+      } = getField(thresholdData, ALERT_THRESHOLD_RESULT);
 
-      try {
```

Comment:
This section was for backward compatibility with a (very) old data format. Since this is a major release, it seems like a good time to remove it.

Reply:
Worst case is they receive extra events in their timeline.